akka-persistence/src/main/scala/akka/persistence/journal/AsyncWriteJournal.scala

```diff
@@ -69,7 +69,8 @@ trait AsyncWriteJournal extends Actor with WriteJournalBase with AsyncRecovery {
         val atomicWriteCount = messages.count(_.isInstanceOf[AtomicWrite])
         val prepared = Try(preparePersistentBatch(messages))
         val writeResult = (prepared match {
-          case Success(prep) =>
+          case Success(prep) if prep.isEmpty => Future.successful(Nil)
+          case Success(prep)                 =>
```

Comment:
Let's see if I understand this correctly.

`messages: immutable.Seq[PersistentEnvelope]`, where PersistentEnvelope can be AtomicWrite or NonPersistentRepr. NonPersistentRepr are used for the defer.

`preparePersistentBatch` collects AtomicWrite, so if messages only contains NonPersistentRepr then `asyncWriteMessages` are called with empty collection.

So if empty then we continue right away instead. The ordering of the replies should be handled by `Resequencer` anyway.

Reply:
Exactly